xbmc/utils/BufferObjectFactory.cpp

```diff
@@ -8,17 +8,22 @@
 
 #include "BufferObjectFactory.h"
 
-std::vector<std::function<std::unique_ptr<CBufferObject>()>> CBufferObjectFactory::m_bufferObjects;
+std::list<std::function<std::unique_ptr<CBufferObject>()>> CBufferObjectFactory::m_bufferObjects;
 
 std::unique_ptr<CBufferObject> CBufferObjectFactory::CreateBufferObject()
 {
-  return m_bufferObjects.back()();
+  for (const auto bufferObject : m_bufferObjects)
+  {
+    return bufferObject();
+  }
```

Comment:
I can sort of see what you did there, but this reads really unintuitively. What's wrong with the old code and adding a `.empty()` check?